### PR TITLE
feat(datetime): add preferWheel support for date-time and time-date presentation

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: Part of https://github.com/ionic-team/ionic-framework/issues/23981

## What is the new behavior?

- Add support for the wheel style when presentation="date-time" or `presentation="time-date"`. Intended behavior can be found on the design doc for this feature.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
I am still working on https://github.com/ionic-team/ionic-framework/pull/25432/files#diff-2096901ec14aa009bc4a017f56dd504533b6e1c2fb939717be4fe18f331f7a9fR457, but everything else should be ready for a review.

File: core/src/components/datetime/test/prefer-wheel/index.html [hunk 1]
```diff
@@ -50,6 +50,14 @@
       locaelInput.onchange = (ev) => {
         datetime.locale = ev.target.value || 'en-US';
       };
+      datetime.isDateEnabled = (dateIsoString) => {
```
Comment (amandaejohnston) [inline core/src/components/datetime/test/prefer-wheel/index.html:53]: I'm not sure if this is due to this PR or something else, but I noticed that the disabled dates aren't grayed out: https://user-images.githubusercontent.com/90629384/173372063-340d126a-b344-4e60-8406-b6f6b290cc16.mp4

Comment (liamdebeasi) [reply core/src/components/datetime/test/prefer-wheel/index.html:53]: Good catch! looks like I wiped the opacity responsible for this during a merge. I re-added the style and I added a screenshot test for this.

**Comment by liamdebeasi** [comment]
I created a ticket to handle https://github.com/ionic-team/ionic-framework/pull/25432/files#diff-2096901ec14aa009bc4a017f56dd504533b6e1c2fb939717be4fe18f331f7a9fR457 as I will not be able to fix it this sprint. I'll prioritize it for one of the upcoming sprints.

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**